client/services/code_preprocessors/PythonCodePreprocessorService.js

```diff
@@ -279,7 +284,7 @@ tie.factory('PythonCodePreprocessorService', [
         codeSubmission.append([
           auxiliaryCode,
           this._generateCorrectnessTestCode(
-            correctnessTests, mainFunctionName, outputFunctionName),
+            correctnessTests, inputFunctionName, mainFunctionName, outputFunctionName),
           this._generateBuggyOutputTestCode(
```

Comment:
No, I think performance tests are separate.

(Btw, it might also be a good idea to write some Karma tests in the corresponding spec file to verify the expected behavior.)